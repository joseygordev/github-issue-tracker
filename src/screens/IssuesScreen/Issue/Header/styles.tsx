import Colors from '../../../../constants/Colors';
import { Dimensions, StyleSheet } from 'react-native';

export default StyleSheet.create({
  card: {
    backgroundColor: Colors.white,
    paddingHorizontal: 16,
    paddingVertical: 16,
    borderRadius: 8,
    marginBottom: 16,
    elevation: 4,
    shadowColor: Colors.steel,
    shadowOffset: {
      width: 2,
      height: 2,
    },
    shadowRadius: 3,
    shadowOpacity: 0.1,
  },
  profileImage: {
    width: 20,
    height: 20,
    borderRadius: 20 / 2,
    marginRight: 8,
    marginTop: 1,
  },
  container: {
    flexDirection: 'column',
  },
  content: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  username: {
    color: Colors.steel,
    fontSize: 12,
  },
  title: {
    maxWidth: Dimensions.get('screen').width - 100,
  },
  labels: {
    flexDirection: 'row',
    marginTop: 8,
    flexWrap: 'wrap',
  },
  label: {
    borderRadius: 16,
    paddingHorizontal: 10,
    height: 22,
    alignItems: 'center',
    flexDirection: 'row',
    marginRight: 8,
    marginTop: 8,
  },
  labelText: {
    fontSize: 12,
  },
  timeAgo: {
    position: 'absolute',
    top: 10,
    right: 14,
  },
  timeAgoText: {
    color: Colors.border,
    fontSize: 12,
  },
  state: {
    paddingRight: 12,
    paddingLeft: 12,
    paddingVertical: 4,
    borderRadius: 16,
    maxWidth: 80,
    marginVertical: 8
  },
  stateText: {
    color: Colors.white,
    textAlign: 'center'
  },
});
