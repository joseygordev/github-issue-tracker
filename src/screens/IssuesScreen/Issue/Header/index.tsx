import React from 'react';
import moment from 'moment';
import { View, Image } from 'react-native';
import { AntDesign } from '@expo/vector-icons';

import { IssueHeaderProps } from '../../../../types/issues';
import Colors from '../../../../constants/Colors';

import CustomText from '../../../../components/CustomText';

import styles from './styles';

export default function Header(props: IssueHeaderProps) {
  const { issue } = props;

  return (
    <View style={styles.container}>
      <View style={styles.content}>
        <Image style={styles.profileImage} source={{ uri: issue.user.avatar_url }} />
        <CustomText style={styles.username}>{issue.user.login}</CustomText>
        <CustomText style={styles.timeAgoText}> • {moment.duration(moment().diff(moment(issue.created_at))).humanize() + ' ago'}</CustomText>
        {issue.isBookmarked && (
          <>
            <CustomText style={styles.timeAgoText}> • </CustomText>
            <AntDesign size={16} color={Colors.border} name="star" />
          </>
        )}
      </View>
      <View style={[styles.state, { backgroundColor: issue.state == 'open' ? Colors.green : Colors.border }]}>
        <CustomText style={styles.stateText}>{issue.state[0].toUpperCase() + issue.state.slice(1)}</CustomText>
      </View>
    </View>
  );
}
