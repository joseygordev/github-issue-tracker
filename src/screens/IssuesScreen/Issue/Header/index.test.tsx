import React from 'react';
import { create, act } from 'react-test-renderer';

import mockIssues from '../../../../data/mocks/issues';
import Header from './index';

describe('Header', () => {  
  const issue = mockIssues[0];

  it('it renders Header correctly', async () => {
    const tree = create(<Header issue={issue} />);

    expect(tree).toMatchSnapshot();
  });
  
  it('it renders Header with the user name', async () => {
    const tree = create(<Header issue={issue} />);
    const headerComponent = tree.root.findByProps({ testID: 'userName' }).props;
  
    expect(headerComponent.children).toEqual(issue.user.login);
  });
  
  it('it renders Header bookmarked', async () => {
    const tree = create(<Header issue={{...issue, isBookmarked: true}} />);
    const headerComponent = tree.root.findByProps({ testID: 'bookmarked' });

    await act(async () => jest.runAllTimers());
  
    expect(headerComponent).toBeTruthy();
  });
});
