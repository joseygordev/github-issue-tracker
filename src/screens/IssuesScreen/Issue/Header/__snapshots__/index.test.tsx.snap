// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header it renders Header correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "column",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "mock_url",
        }
      }
      style={
        Object {
          "borderRadius": 10,
          "height": 20,
          "marginRight": 8,
          "marginTop": 1,
          "width": 20,
        }
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "#000",
          },
          Object {
            "color": "#414d5b",
            "fontSize": 12,
          },
          Object {
            "fontFamily": "space-mono",
          },
        ]
      }
      testID="userName"
    >
      mock_login
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#000",
          },
          Object {
            "color": "#aab2ba",
            "fontSize": 12,
          },
          Object {
            "fontFamily": "space-mono",
          },
        ]
      }
      testID="textValue"
    >
       â€¢ 
      a few seconds ago
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "borderRadius": 16,
          "marginVertical": 8,
          "maxWidth": 80,
          "paddingLeft": 12,
          "paddingRight": 12,
          "paddingVertical": 4,
        },
        Object {
          "backgroundColor": "#16c65d",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000",
          },
          Object {
            "color": "#fff",
            "textAlign": "center",
          },
          Object {
            "fontFamily": "space-mono",
          },
        ]
      }
      testID="textValue"
    >
      Open
    </Text>
  </View>
</View>
`;
